Applications/Spire/Source/Ui/QuantityBox.cpp

```diff
@@ -14,14 +16,65 @@ namespace {
   };
 }
 
+struct QuantityBox::UnsignedQuantityModel : OptionalQuantityModel {
+  mutable CurrentSignal m_current_signal;
+  std::shared_ptr<OptionalQuantityModel> m_model;
+  scoped_connection m_current_connection;
+
+  UnsignedQuantityModel(std::shared_ptr<OptionalQuantityModel> model)
+    : m_model(std::move(model)),
+      m_current_connection(m_model->connect_current_signal(
+        [=] (const auto& current) {
+          on_current(current);
+        })) {}
+
+  optional<Quantity> get_minimum() const override {
+    return 0;
+  }
+
+  optional<Quantity> get_maximum() const override {
+    return m_model->get_maximum();
+  }
+
+  Quantity get_increment() const override {
+    return m_model->get_increment();
+  }
+
+  QValidator::State get_state() const override {
+    return m_model->get_state();
+  }
+
+  const optional<Quantity>& get_current() const override {
+    return m_model->get_current();
+  }
+
+  QValidator::State set_current(const optional<Quantity>& value) override {
+    if(value && *value < 0) {
```

Comment:
Never suppress errors, all this does is hide an issue and makes it incredibly hard to debug and identify the cause of a problem. When a precondition is violated, use the most appropriate mechanism available to report the issue as soon as possible so it can be handled locally.